### ISSUE TITLE
Error when /api/v1/settings/notifications is not found

### ISSUE THREAD

**Issue by Flole998** [openhab/openhab-android#2358]
This is an issue that is related to openhab/openhab-core#2079. While the other issue covers unexpected behavior on the server this one is for unexpected behavior in the App.

When using a local openHAB Server the App requests /api/v1/settings/notifications, which is for myopenHAB and doesn't exist on a local installation. This causes the App to display a message at the bottom (. We should either remove this call (at least when contacting a local server) or silence the warning (at least when contacting a local server). I am not sure if it isn't trying to use my local server or if it is also trying to request this on a local server (which it shouldn't in my opinion). I would completely silence this warning for this URL as some might have the local server configured with the local address and the remote server configured with an external IP Address.

**Comment by maniac103** [comment]
Did you enter the URL of your local server as 'remote server' in the app? The app will only do this request on the remote server.

**Comment by Flole998** [comment]
I entered a locally reachable URL for local server and a remotely reachable URL for the remote URL. It should be able to contact the local URL in my case, but seems like it doesn't.

Also I can't remove the remote URL anymore, if I empty the field it doesn't seem to save it.

**Comment by maniac103** [comment]
Please post the app log after a connection attempt.

**Comment by Flole998** [comment]
Seems like it is actually using the local connection?

```
--------- beginning of main
01-09 19:35:37.860 10434 10434 I AppCompatViewInflater: app:theme is now deprecated. Please move to using android:theme instead.
01-09 19:35:37.890 10434 10434 I MediaPlayer: constructor
01-09 19:35:37.975 10434 10434 D Ringtone: Successfully created local player
01-09 19:35:37.991 10434 10434 D PreferencesActivity: Removing notification prefs for < 25
01-09 19:35:38.155 10434 10434 D OnePlusJankManager:  Chor uploadMDM JANK_TYPE_ONCE mViewTitle = org.openhab.habdroid.beta/org.openhab.habdroid.ui.MainActivity--- jank level = 1
01-09 19:35:38.207 10434 10434 D OnePlusJankManager:  Chor uploadMDM JANK_TYPE_ONCE mViewTitle = org.openhab.habdroid.beta/org.openhab.habdroid.ui.PreferencesActivity--- jank level = 1
01-09 19:35:38.615 10434 10434 W ListPreference: Setting a summary with a String formatting marker is no longer supported. You should use a SummaryProvider instead.
01-09 19:35:38.654 10434 10434 D MainActivity: onStop()
01-09 19:35:38.654 10434 10434 D PageConnectionHolderFragment: onStop()
01-09 19:35:38.654 10434 10434 D PageConnectionHolderFragment: Canceling connection for URL https://<openhab-local-address-Flole>/rest/sitemaps/default/default
01-09 19:35:38.655 10434 10434 D MainActivity: onSaveInstanceState()
01-09 19:35:38.655 10434 10434 D ContentController: onSaveInstanceState()
01-09 19:35:39.774 10434 10434 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@38526c5
01-09 19:35:39.790 10434 10434 I AppCompatViewInflater: app:theme is now deprecated. Please move to using android:theme instead.
01-09 19:35:39.792 10434 10434 I FloatingActionButton: Setting a custom background is not supported.
01-09 19:35:39.832 10434 10434 D LogActivity: onCreateOptionsMenu()
01-09 19:35:40.641 10434 10434 D OnePlusJankManager:  Chor uploadMDM JANK_TYPE_NODRAW mViewTitle = org.openhab.habdroid.beta/org.openhab.habdroid.ui.LogActivity--- jank level = 1
01-09 19:35:40.845 10434 10434 D LogActivity: onOptionsItemSelected()
--------- beginning of events
01-09 19:35:41.667 10434 10434 I am_on_paused_called: [0,org.openhab.habdroid.ui.LogActivity,performPause]
01-09 19:35:41.698 10434 10434 I am_on_restart_called: [0,org.openhab.habdroid.ui.PreferencesActivity,performRestartActivity]
01-09 19:35:41.718 10434 10434 I am_on_start_called: [0,org.openhab.habdroid.ui.PreferencesActivity,handleStartActivity]
01-09 19:35:41.720 10434 10434 I am_on_resume_called: [0,org.openhab.habdroid.ui.PreferencesActivity,RESUME_ACTIVITY]
01-09 19:35:41.746 10434 19992 D ConnectionFactory: checkAvailableConnection: found types [ConnectionType(type = Wifi, network=113)]
01-09 19:35:41.746 10434 19992 D AbstractConnection: Checking reachability of https://<openhab-local-address-Flole>/ (via 113)
01-09 19:35:41.772 10434 19992 D AbstractConnection: Socket connected (attempt  0)
01-09 19:35:41.773 10434 19992 D ConnectionFactory: Connecting to local URL via ConnectionType(type = Wifi, network=113)
01-09 19:35:42.194 10434 10434 I am_on_stop_called: [0,org.openhab.habdroid.ui.LogActivity,LIFECYCLER_STOP_ACTIVITY]
01-09 19:35:42.195 10434 10434 I am_on_destroy_called: [0,org.openhab.habdroid.ui.LogActivity,performDestroy]
01-09 19:35:42.356 10434 10434 I am_on_paused_called: [0,org.openhab.habdroid.ui.PreferencesActivity,performPause]
01-09 19:35:42.373 10434 10434 D MainActivity: onActivityResult() requestCode = 1001, resultCode = -1
01-09 19:35:42.373 10434 10434 I am_on_activity_result_called: [0,org.openhab.habdroid.ui.MainActivity,ACTIVITY_RESULT]
01-09 19:35:42.374 10434 10434 I am_on_restart_called: [0,org.openhab.habdroid.ui.MainActivity,performRestartActivity]
01-09 19:35:42.374 10434 10434 D MainActivity: onStart()
01-09 19:35:42.374 10434 10434 D PageConnectionHolderFragment: onStart(), started false
01-09 19:35:42.374 10434 10434 D PageConnectionHolderFragment: Loading data for https://<openhab-local-address-Flole>/rest/sitemaps/default/default, long polling false
01-09 19:35:42.375 10434 10434 D WidgetListFragment: onStart() https://<openhab-local-address-Flole>/rest/sitemaps/default/default
01-09 19:35:42.375 10434 10434 D PageConnectionHolderFragment: Trigger update for URL https://<openhab-local-address-Flole>/rest/sitemaps/default/default, force false
01-09 19:35:42.375 10434 10434 D ContentController: Got update for URL https://<openhab-local-address-Flole>/rest/sitemaps/default/default, pending []
01-09 19:35:42.385 10434 10434 D MainActivity: onActiveConnectionChanged()
01-09 19:35:42.392 10434 10434 D MainActivity: Cannot auto select server: No server with configured wifi
01-09 19:35:42.392 10434 10434 I am_on_start_called: [0,org.openhab.habdroid.ui.MainActivity,handleStartActivity]
01-09 19:35:42.392 10434 10434 D MainActivity: onResume()
01-09 19:35:42.398 10434 10434 I am_on_resume_called: [0,org.openhab.habdroid.ui.MainActivity,RESUME_ACTIVITY]
01-09 19:35:42.860 10434 10434 I am_on_stop_called: [0,org.openhab.habdroid.ui.PreferencesActivity,LIFECYCLER_STOP_ACTIVITY]
01-09 19:35:42.863 10434 10434 I am_on_destroy_called: [0,org.openhab.habdroid.ui.PreferencesActivity,performDestroy]
01-09 19:35:42.925 10434 10434 D MainActivity: onBackPressed()
01-09 19:35:42.940 10434 10434 D MainActivity: onPause()
01-09 19:35:42.940 10434 10434 D WidgetListFragment: onPause() https://<openhab-local-address-Flole>/rest/sitemaps/default/default
01-09 19:35:42.942 10434 10434 I am_on_paused_called: [0,org.openhab.habdroid.ui.MainActivity,performPause]
01-09 19:35:43.036 10434 10434 D PageConnectionHolderFragment: Updated page data for URL https://<openhab-local-address-Flole>/rest/sitemaps/default/default (18 widgets)
01-09 19:35:43.036 10434 10434 D ContentController: Got update for URL https://<openhab-local-address-Flole>/rest/sitemaps/default/default, pending []
01-09 19:35:43.119 10434 10434 W StaticLayout: maxLineHeight should not be -1.  maxLines:1 lineCount:1
01-09 19:35:43.161 10434 10452 I b.habdroid.bet: Background concurrent copying GC freed 73936(3MB) AllocSpace objects, 37(868KB) LOS objects, 49% free, 5MB/10MB, paused 112us total 144.963ms
01-09 19:35:43.187 10434 10449 V MediaPlayer: resetDrmState:  mDrmInfo=null mDrmProvisioningThread=null mPrepareDrmInProgress=false mActiveDrmScheme=false
01-09 19:35:43.187 10434 10449 V MediaPlayer: cleanDrmObj: mDrmObj=null mDrmSessionId=null
01-09 19:35:43.353 10434 10434 D MainActivity: onStop()
01-09 19:35:43.354 10434 10434 D PageConnectionHolderFragment: onStop()
01-09 19:35:43.354 10434 10434 D PageConnectionHolderFragment: Canceling connection for URL https://<openhab-local-address-Flole>/rest/sitemaps/default/default
01-09 19:35:43.354 10434 10434 I am_on_stop_called: [0,org.openhab.habdroid.ui.MainActivity,LIFECYCLER_STOP_ACTIVITY]
01-09 19:35:43.359 10434 10434 I am_on_destroy_called: [0,org.openhab.habdroid.ui.MainActivity,performDestroy]
01-09 19:35:44.599 10434 10434 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@57f2ca4
01-09 19:35:44.635 10434 10434 D MainActivity: onCreate()
01-09 19:35:44.673 10434 10434 I AppCompatViewInflater: app:theme is now deprecated. Please move to using android:theme instead.
01-09 19:35:44.694 10434 10434 D MainActivity: Got intent: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=org.openhab.habdroid.beta/org.openhab.habdroid.ui.MainActivity bnds=[237,200][439,487] }
01-09 19:35:44.696 10434 10434 I am_on_create_called: [0,org.openhab.habdroid.ui.MainActivity,performCreate]
01-09 19:35:44.698 10434 10434 D MainActivity: onStart()
01-09 19:35:44.698 10434 10434 D PageConnectionHolderFragment: onStart(), started false
01-09 19:35:44.707 10434 10434 D MainActivity: onActiveConnectionChanged()
01-09 19:35:44.720 10434 10434 D AbstractBaseActivity: Queue snackbar with tag connectionEstablished
01-09 19:35:44.720 10434 10434 D BackgroundTasksManager: triggerPeriodicWork()
01-09 19:35:44.721 10434 10434 D BackgroundTasksManager: Periodic workers are not needed, canceling...
01-09 19:35:44.721 10434 10434 I chatty  : uid=10175(org.openhab.habdroid.beta) identical 2 lines
01-09 19:35:44.721 10434 10434 D BackgroundTasksManager: Periodic workers are not needed, canceling...
01-09 19:35:44.721 10434 10434 D ContentController: Update to connection org.openhab.habdroid.core.connection.DefaultConnection@30a45b8 (message null)
01-09 19:35:44.721 10434 10434 D PageConnectionHolderFragment: updateActiveConnections: URL list [], connection org.openhab.habdroid.core.connection.DefaultConnection@30a45b8
01-09 19:35:44.734 10434 20029 D AbstractBaseActivity: Queue snackbar with tag pushNotificationFail
01-09 19:35:44.735 10434 20029 D AbstractBaseActivity: No next snackbar to show
01-09 19:35:44.752 10434 10434 D ContentController: clearServerCommunicationFailure()
01-09 19:35:44.752 10434 10434 D BackgroundTasksManager: triggerPeriodicWork()
01-09 19:35:44.752 10434 10434 D BackgroundTasksManager: Periodic workers are not needed, canceling...
01-09 19:35:44.752 10434 10434 I chatty  : uid=10175(org.openhab.habdroid.beta) identical 2 lines
01-09 19:35:44.752 10434 10434 D BackgroundTasksManager: Periodic workers are not needed, canceling...
01-09 19:35:44.757 10434 10434 D MainActivity: Cannot auto select server: No server with configured wifi
01-09 19:35:44.757 10434 10434 I am_on_start_called: [0,org.openhab.habdroid.ui.MainActivity,handleStartActivity]
01-09 19:35:44.757 10434 10434 D MainActivity: onPostCreate()
01-09 19:35:44.757 10434 10434 D MainActivity: onResume()
01-09 19:35:44.761 10434 10434 I am_on_resume_called: [0,org.openhab.habdroid.ui.MainActivity,RESUME_ACTIVITY]
01-09 19:35:44.770 10434 20029 D ConnectionFactory: checkAvailableConnection: found types [ConnectionType(type = Wifi, network=113)]
01-09 19:35:44.770 10434 20029 D AbstractConnection: Checking reachability of https://<openhab-local-address-Flole>/ (via 113)
01-09 19:35:44.784 10434 20029 D AbstractConnection: Socket connected (attempt  0)
01-09 19:35:44.785 10434 20029 D ConnectionFactory: Connecting to local URL via ConnectionType(type = Wifi, network=113)
01-09 19:35:44.808 10434 10434 D MainActivity: onCreateOptionsMenu()
01-09 19:35:44.810 10434 10434 D MainActivity: onPrepareOptionsMenu()
01-09 19:35:44.837 10434 10434 D MainActivity: onActiveCloudConnectionChanged()
01-09 19:35:44.844 10434 10434 D MainActivity: onPrimaryCloudConnectionChanged()
01-09 19:35:44.860 10434 20029 D AbstractBaseActivity: Remove snackbar with tag pushNotificationFail from queue
01-09 19:35:44.861 10434 20029 D AbstractBaseActivity: Queue snackbar with tag pushNotificationFail
01-09 19:35:44.861 10434 20029 D AbstractBaseActivity: No next snackbar to show
01-09 19:35:44.974 10434 10434 D AbstractBaseActivity: Show snackbar with tag connectionEstablished
01-09 19:35:45.681 10434 10434 I ServerProperties: Server has rest api version 4
01-09 19:35:46.292 10434 10434 D ServerProperties: Server returned sitemaps: [Sitemap(name=fvs, label=FvS, icon=null, homepageLink=https://<openhab-local-address-Flole>/rest/sitemaps/fvs/fvs), Sitemap(name=default, label=Hauptmenü, icon=null, homepageLink=https://<openhab-local-address-Flole>/rest/sitemaps/default/default)]
01-09 19:35:46.295 10434 10434 D MainActivity: Configured sitemap is 'default', selected Sitemap(name=default, label=Hauptmenü, icon=null, homepageLink=https://<openhab-local-address-Flole>/rest/sitemaps/default/default)
01-09 19:35:46.296 10434 10434 D ContentController: Opening sitemap Sitemap(name=default, label=Hauptmenü, icon=null, homepageLink=https://<openhab-local-address-Flole>/rest/sitemaps/default/default) (current: null)
01-09 19:35:46.296 10434 10434 D PageConnectionHolderFragment: updateActiveConnections: URL list [], connection org.openhab.habdroid.core.connection.DefaultConnection@30a45b8
01-09 19:35:46.296 10434 10434 D PageConnectionHolderFragment: updateActiveConnections: URL list [https://<openhab-local-address-Flole>/rest/sitemaps/default/default], connection org.openhab.habdroid.core.connection.DefaultConnection@30a45b8
01-09 19:35:46.296 10434 10434 D PageConnectionHolderFragment: Creating new handler for URL https://<openhab-local-address-Flole>/rest/sitemaps/default/default
01-09 19:35:46.297 10434 10434 D PageConnectionHolderFragment: Creating new SSE helper for sitemap default, page default
01-09 19:35:46.297 10434 10434 D PageConnectionHolderFragment: Loading data for https://<openhab-local-address-Flole>/rest/sitemaps/default/default, long polling false
01-09 19:35:46.614 10434 10434 W StaticLayout: maxLineHeight should not be -1.  maxLines:2 lineCount:2
01-09 19:35:46.620 10434 10434 I chatty  : uid=10175(org.openhab.habdroid.beta) identical 8 lines
01-09 19:35:46.620 10434 10434 W StaticLayout: maxLineHeight should not be -1.  maxLines:2 lineCount:2
01-09 19:35:46.781 10434 10434 D AbstractBaseActivity: Show snackbar with tag pushNotificationFail
01-09 19:35:47.129 10434 10434 D PageConnectionHolderFragment: Updated page data for URL https://<openhab-local-address-Flole>/rest/sitemaps/default/default (18 widgets)
01-09 19:35:47.130 10434 10434 D ContentController: Got update for URL https://<openhab-local-address-Flole>/rest/sitemaps/default/default, pending [https://<openhab-local-address-Flole>/rest/sitemaps/default/default]
01-09 19:35:47.169 10434 10434 D WidgetListFragment: onViewCreated() https://<openhab-local-address-Flole>/rest/sitemaps/default/default
01-09 19:35:47.170 10434 10434 D WidgetListFragment: onStart() https://<openhab-local-address-Flole>/rest/sitemaps/default/default
01-09 19:35:47.170 10434 10434 D PageConnectionHolderFragment: Trigger update for URL https://<openhab-local-address-Flole>/rest/sitemaps/default/default, force false
01-09 19:35:47.170 10434 10434 D ContentController: Got update for URL https://<openhab-local-address-Flole>/rest/sitemaps/default/default, pending []
01-09 19:35:47.522 10434 10434 D MainActivity: onOptionsItemSelected()
01-09 19:35:48.180 10434 10434 E b.habdroid.bet: No package ID 5c found for ID 0x5c13d641.
01-09 19:35:49.070 10434 10434 D MainActivity: onPause()
01-09 19:35:49.072 10434 10434 D WidgetListFragment: onPause() https://<openhab-local-address-Flole>/rest/sitemaps/default/default
01-09 19:35:49.074 10434 10434 I am_on_paused_called: [0,org.openhab.habdroid.ui.MainActivity,performPause]
01-09 19:35:49.091 10434 10434 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@2afb623
01-09 19:35:49.105 10434 10434 I AppCompatViewInflater: app:theme is now deprecated. Please move to using android:theme instead.
01-09 19:35:49.107 10434 10434 I am_on_create_called: [0,org.openhab.habdroid.ui.PreferencesActivity,performCreate]
01-09 19:35:49.120 10434 10434 I MediaPlayer: constructor
01-09 19:35:49.145 10434 10434 D Ringtone: Successfully created local player
01-09 19:35:49.154 10434 10434 D PreferencesActivity: Removing notification prefs for < 25
01-09 19:35:49.165 10434 10434 I am_on_start_called: [0,org.openhab.habdroid.ui.PreferencesActivity,handleStartActivity]
01-09 19:35:49.166 10434 10434 I am_on_resume_called: [0,org.openhab.habdroid.ui.PreferencesActivity,RESUME_ACTIVITY]
01-09 19:35:49.624 10434 10434 D AbstractBaseActivity: No next snackbar to show
01-09 19:35:49.709 10434 10434 D MainActivity: onStop()
01-09 19:35:49.709 10434 10434 D PageConnectionHolderFragment: onStop()
01-09 19:35:49.709 10434 10434 D PageConnectionHolderFragment: Canceling connection for URL https://<openhab-local-address-Flole>/rest/sitemaps/default/default
01-09 19:35:49.709 10434 10434 I am_on_stop_called: [0,org.openhab.habdroid.ui.MainActivity,STOP_ACTIVITY_ITEM]
01-09 19:35:49.710 10434 10434 D MainActivity: onSaveInstanceState()
01-09 19:35:49.710 10434 10434 D ContentController: onSaveInstanceState()
01-09 19:35:49.952 10434 10434 W ListPreference: Setting a summary with a String formatting marker is no longer supported. You should use a SummaryProvider instead.
01-09 19:35:50.966 10434 10434 I am_on_paused_called: [0,org.openhab.habdroid.ui.PreferencesActivity,performPause]
01-09 19:35:50.985 10434 10434 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@f94536
01-09 19:35:51.003 10434 10434 I AppCompatViewInflater: app:theme is now deprecated. Please move to using android:theme instead.
01-09 19:35:51.005 10434 10434 I FloatingActionButton: Setting a custom background is not supported.
01-09 19:35:51.019 10434 10434 I am_on_create_called: [0,org.openhab.habdroid.ui.LogActivity,performCreate]
01-09 19:35:51.021 10434 10434 I am_on_start_called: [0,org.openhab.habdroid.ui.LogActivity,handleStartActivity]
01-09 19:35:51.022 10434 10434 I am_on_resume_called: [0,org.openhab.habdroid.ui.LogActivity,RESUME_ACTIVITY]
01-09 19:35:51.070 10434 10434 D LogActivity: onCreateOptionsMenu()
01-09 19:35:51.082 10434 10449 W MediaPlayer-JNI: MediaPlayer finalized without being released
01-09 19:35:51.093 10434 10449 V MediaPlayer: resetDrmState:  mDrmInfo=null mDrmProvisioningThread=null mPrepareDrmInProgress=false mActiveDrmScheme=false
01-09 19:35:51.093 10434 10449 V MediaPlayer: cleanDrmObj: mDrmObj=null mDrmSessionId=null
```

**Comment by Flole998** [comment]
To add to this: This is on the latest Beta (2.15.15-beta), the latest Release (2.15.0) works perfectly fine.

**Comment by maniac103** [comment]
This log shows it's correctly using the local connection. If a remote connection is set, it's still correct the app tries to connect to the notification endpoint on the remote server in that case, since the app can't know whether the remote server is just an OH server or an openhab-cloud instance.

**Comment by Flole998** [comment]
That shouldn't cause a visible error though.

It says something like "Not available, error while trying to connect to the server for remote connection: Not found (H...". The German version is "Nicht verfügbar, Fehler beim Verbinden mit dem Server für den Fernzugriff: Nicht gefunden (H...".

**Comment by maniac103** [comment]
Yes, that's right. We should remove the snackbar if the error was a 404.

**Comment by maniac103** [comment]
@mueller-ma On a semi related topic, I'm often getting this recently: 
![IMG_20210109_204754](https://user-images.githubusercontent.com/781373/104107413-47fca100-52bc-11eb-9057-6b1ac7cb5ec1.jpg)
Relevant log entry for that:
```
01-09 20:52:53.267 30957 31034 E Util    : REST call to  failed
01-09 20:52:53.267 30957 31034 E Util    : kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelled}@424bb37
01-09 20:52:53.268 30957 31034 D AbstractBaseActivity: Remove snackbar with tag pushNotificationFail from queue
01-09 20:52:53.268 30957 31034 D AbstractBaseActivity: Queue snackbar with tag pushNotificationFail
```

I assume it's two successive connection updates for whatever reason, where the second cancels the first one. We should exempt CancellationException from showing a snackbar as well.

**Comment by mueller-ma** [comment]
I see a couple of issues here and most of them have already been fixed, but we cannot release new versions to Play Store currently (working on it!) See #2336

- Remote server is detected as cloud server and shows an error: Fixed by #2325. I just checked the code: No snackbar is shown anymore in this case.
- The error message doesn't refer to notifications: Fixed by #2330
- Url is settings cannot be removed: Fixed by  #2324
- JobCancellationException: Not fixed yet.
